### PR TITLE
feat: set the correct external sender key when creating MLS conversation

### PR DESCRIPTION
**Pull request by MohamadJaara**
----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [ ] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [ ] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

### Issues

when creating a new conversation, we need to set the server public key matching the default cipher suite

### Solutions

select the matching signature from cipher suite

Needs releases with:

- #2737 
- #2740 

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/mlspublickeys/MLSPublicKeysMapper.kt [hunk 1]
```diff
@@ -19,25 +19,15 @@
...
     fun toCrypto(publicKey: MLSPublicKey): com.wire.kalium.cryptography.Ed22519Key
     fun toCrypto(externalSenderKey: ExternalSenderKey): com.wire.kalium.cryptography.Ed22519Key
+    fun fromCipherSuite(cipherSuite: CipherSuite): MLSPublicKeyType
 }
 
 class MLSPublicKeysMapperImpl : MLSPublicKeysMapper {
-    override fun fromDTO(publicKeys: MLSPublicKeysDTO) = with(publicKeys) {
-        removal?.entries?.mapNotNull {
-            when (it.key) {
-                ED25519 -> MLSPublicKey(Ed25519Key(it.value.decodeBase64Bytes()), KeyType.REMOVAL)
```
Comment (MohamadJaara) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/mlspublickeys/MLSPublicKeysMapper.kt:35]: this mapper will always return a list containing one item or null, so it was removed with choosing the correct signature from the map

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/data/mlspublickeys/MLSPublicKeysRepository.kt [hunk 2]
```diff
@@ -19,34 +19,61 @@
...
-            keys
+            MLSPublicKeys(removal = it.removal)
         }
 
-    override suspend fun getKeys(): Either<CoreFailure, List<MLSPublicKey>> {
+    override suspend fun getKeys(): Either<CoreFailure, MLSPublicKeys> {
         return publicKeys?.let { Either.Right(it) } ?: fetchKeys()
     }
 
+    override suspend fun keyForCipherSuite(cipherSuite: CipherSuite): Either<CoreFailure, ByteArray> {
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/data/mlspublickeys/MLSPublicKeysRepository.kt:63]:
```suggestion
    override suspend fun getKeyForCipherSuite(cipherSuite: CipherSuite): Either<CoreFailure, ByteArray> {
```

File: logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterMLSClientUseCase.kt [hunk 3]
```diff
@@ -53,23 +54,33 @@ internal class RegisterMLSClientUseCaseImpl(
...
+        return userConfigRepository.getE2EISettings().flatMap { e2eiSettings ->
+            kaliumLogger.d("Registering MLS client with E2EI settings")
             if (e2eiSettings.isRequired && !mlsClientProvider.isMLSClientInitialised()) {
+                kaliumLogger.d("E2EI settings are required but MLS client is not initialised")
                 return RegisterMLSClientResult.E2EICertificateRequired.right()
             } else {
                 mlsClientProvider.getMLSClient(clientId)
             }
         }.onFailure {
+            kaliumLogger.e("Failed to get MLS client: $it")
```
Comment (mchenani) [inline logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/client/RegisterMLSClientUseCase.kt:68]: this is a wrong error, this case happens when e2ei is disabled or can't be fetched, then we proceed with normal mls client registeration flow.

**Review by mchenani — APPROVED** [review]
LGTM, added couple of comments.

**Review by vitorhugods — APPROVED** [review]
Tiny nitpick

**Pull request merged**